techmatters/flex-plugins
# CHI-2431: chat transfer fix

## Description

* Added a HRM contact ID to tasks now we create them up front
* Moved transfer receiving logic to be part of the contact creation action, and able to resume from partially completes states on refresh

### Checklist
- N/A Feature flags added
- N/A Strings are localized

### Related Issues
CHI-2431

### Verification steps

Difficult - you need to add code to simulate accepting chat transfers ending partway through in 2 stages:

- The ownership is never updated in HRM
- The ownership is updated, but task control is never moved to the receiving task (chat transfers only)

Then you need to set the code back to the checked in version and refresh, the transfer should complete and leave the receivers UI in a state where the contact can be completed & chat can continue

Also regression cold & warm voice transfers
